### DISCUSSION TITLE
Custom link HREF empty

### DISCUSSION THREAD

**Discussion by jkoopmann** [netbox-community/netbox#17949] [Help Wanted!] [answered]
### Deployment Type

Self-hosted

### Triage priority

N/A

### NetBox Version

v4.1.6

### Python Version

3.12.3

### Steps to Reproduce

1. Create a custom link for dcim.device
2. Link text: "Push configuration"
3. Link URL: anything. else. https://www.heise.de

### Expected Behavior

When clicking the button it should open the webpage. 

### Observed Behavior

The href seems to be empty:

`<a href="" target="_blank" class="btn btn-sm btn-outline-secondary">Push Configuration</a>`

**Comment by jeremystretch** [comment]
This works as expected. Check your [`ALLOWED_URL_SCHEMES`](https://netbox.readthedocs.io/en/stable/configuration/security/#allowed_url_schemes) to ensure that the `https` scheme has not been mistakenly omitted.

**Comment by jkoopmann** [comment]
Hi @jeremystretch and thanks.

However: It was previously set to *. Now I have

Allowed URL schemes	('http','https','telnet','ssh','vnc')

but still no luck. Still empty refs. This netbox is running behind a HA-proxy. All sorts forwarded-headers are provided, esp.

http-request set-header X-Forwarded-Proto https

What else could I be looking for?

**Reply by jeremystretch** [reply, marked as answer]
`ALLOWED_URL_SCHEMES` is used solely for the sanitization of user-submitted hyperlinks; it has nothing to do with how NetBox itself is running. Try removing the config parameter (restoring the default) and restarting NetBox.

**Comment by jkoopmann** [comment]
Thanks setting it back shows the default is

file, ftp, ftps, http, https, irc, mailto, sftp, ssh, tel, telnet, tftp, vnc, xmpp

so without ('') around it (which I found in the documents but maybe they just very successfully confused me. I was aware of what the parameter was supposed to do. Just that this discussion suggested that it will only work with correct X-Forwarded- headers (which does not make a lot of sense to be but hey....): https://github.com/netbox-community/netbox/discussions/6862


It is working now! Thanks!

**Comment by candlerb** [comment]
> Link URL: anything. else. https://www.heise.de/

The link URL must be *just* a URL. No "anything. else. " in front of it.

**Reply by jkoopmann** [reply]
Ii am not entirely sure whether or not this was meant to be a funny remark dear @candlerb . :-) 

I am fully aware that a link URL is without "anything. else." It was meant to be understood as "whatever I put there, it does not work". And @jeremystretch thankfully pointed me to the correct ALLOWED_URL_SCHEMES which was the solution. So everything is solved now as could be seen at the "accepted answer" status above.

**Reply by candlerb** [reply]
No, it was not meant to be funny. I took it to mean you had put some arbitrary text in front of the URL, like a comment. But I'm glad it's working now.
